spec/unit/actions/synchronize_users_spec.rb

```diff
@@ -44,8 +48,25 @@
         GoodData::LCM2.convert_to_smart_hash(params)
       end
 
-      it_behaves_like 'a user action reading client_id' do
-        let(:client_id) { '123456789' }
+      context 'when the input set is empty' do
+        before do
+          allow(File).to receive(:open).and_return("")
+        end
+        it 'does not fail and logs a warning' do
+          expect(logger).to receive(:warn).with(/not match any client ids/)
+          expect { subject.class.call(params) }.to_not raise_error
+        end
+      end
+
+      context 'when the input set does not contain data for the current project' do
```

Comment:
Similar code found in synchronize_users_spec.rb:51, synchronize_users_spec.rb:61 (mass = 64)